Flaky: outbox backlog drain fencing test asserts an ordering it does not enforce

## Summary

`outboxBacklogDrain.integration.test.ts` fails intermittently on `main`. The
failing case is:

> outbox backlog drain under slow deliveries › when a delivery outlives its
> lease and a rival completes the message › reports the superseded
> acknowledgement as fenced and leaves the row untouched

Observed on `main` at `c3d07af` in the `test-integration` job:

```
AssertionError: expected [] to deeply equal [ 'match-000' ]

- Expected
+ Received

- [
-   "match-000",
- ]
+ []

 ❯ .../outboxBacklogDrain.integration.test.ts:196:33
    196|       expect(slowReport.fenced).toEqual(["match-000"]);

 Test Files  1 failed | 168 passed | 8 skipped (177)
      Tests  1 failed | 1515 passed | 83 skipped (1599)
   Duration  272.01s
```

The assertion one line earlier — `expect(fastReport.dispatched).toEqual(["match-000"])` —
passed, so the rival dispatcher did take the lapsed lease and deliver the
message. Only the slow dispatcher's report came back empty.

## Why it is timing-dependent

The test starts the slow dispatcher without awaiting anything, sleeps 150ms so
its 100ms lease lapses, and then runs the rival:

```ts
const slowRun = slow.runOnce({ now: Date.now() });   // not awaited
await sleep(150);
const fastReport = await fast.runOnce({ now: Date.now() });
```

Nothing synchronises on "the slow dispatcher is actually holding the lease"
before the rival is allowed to run. Two interleavings produce `fenced: []`, and
both are ordinary runner load rather than a defect in the dispatcher:

1. **The slow dispatcher leases nothing.** If its lease query reaches the
   database after the rival has already completed the message, there is no
   pending row to lease. `runOnce` initialises every report array empty and the
   dispatch loop never runs, so the report is `{dispatched: [], fenced: [], …}`.

2. **The slow dispatcher leases the message and then sheds it un-attempted.**
   `dispatchOrShed` releases a message when the remaining lease is below the
   safety margin:

   ```ts
   const elapsedMs = this.clock() - leaseStartedAt;
   if (this.leaseDurationMs - elapsedMs < this.leaseSafetyMarginMs) {
     await this.releaseForLeaseBudget({ message, now, elapsedMs, report });
     return;
   }
   ```

   The margin is 20% of the lease (`LEASE_SAFETY_MARGIN_FRACTION = 0.2`). With
   the test's `leaseDurationMs: 100`, the slow dispatcher has an **80ms** budget
   to get from leasing to the handler. Past that it lands in `report.released`,
   the gated handler is never awaited, and `fenced` stays empty.

Mechanism 2 is the more likely one: shedding is deliberate behaviour introduced
by the same hardening work that added this test, and 80ms is a tight budget on a
shard running 1599 tests in 272s.

Either way the test asserts an ordering it does not enforce.

## Suggested fix

Replace the `sleep(150)` handshake with an explicit wait for the slow
dispatcher to hold the lease before the rival runs — e.g. have the gated handler
signal on entry, await that signal, then advance time past the lease. That
removes the dependency on how quickly the first `runOnce` reaches the database.

Widening `leaseDurationMs` in the test would also enlarge the shed window, but
it only makes the race rarer rather than removing it.

## Not reproduced locally

This analysis is read from the implementation, not from a local repro. The
integration harness starts a ClickHouse testcontainer, which would not start in
the environment where this was investigated, so the failure was not observed
directly.

## Impact

Red `test-integration` runs on `main` and on every branch cut from it, with no
relation to the change under test. Confirmed unrelated to the governance work
that surfaced it.

--- a/platform/app/src/server/event-sourcing/process-manager/outbox/__tests__/outboxBacklogDrain.integration.test.ts
+++ b/platform/app/src/server/event-sourcing/process-manager/outbox/__tests__/outboxBacklogDrain.integration.test.ts
@@ -169,26 +169,51 @@ describe("outbox backlog drain under slow deliveries", () => {
       const result = await store.commit(commit(messages(1)));
       expect(result.outcome).toBe("committed");
 
+      // Two facts have to hold before the rival is allowed to run: the slow
+      // delivery holds the lease, and it is already past the lease-budget
+      // guard so it cannot be shed un-attempted. Waiting for its handler to be
+      // entered establishes both, because the guard runs before the handler
+      // does. Sleeping only hopes for both, and on a loaded runner the slow
+      // dispatcher can lease nothing at all or shed the message on the way to
+      // the handler, either of which reports an empty `fenced`.
+      let announceDeliveryStarted!: () => void;
+      const deliveryStarted = new Promise<void>((resolve) => {
+        announceDeliveryStarted = resolve;
+      });
       let releaseGate!: () => void;
       const gate = new Promise<void>((resolve) => {
         releaseGate = resolve;
       });
+      // Wide enough that the slow dispatcher cannot plausibly run out of lease
+      // budget between leasing and its handler. Nothing waits for this to
+      // expire: the lapse is produced by advancing the rival's clock.
+      const leaseDurationMs = 5_000;
       const slow = new OutboxDispatcherService({
         store,
         processNames: [processName],
-        leaseDurationMs: 100,
-        handlers: { "test.persist": () => gate },
+        leaseDurationMs,
+        handlers: {
+          "test.persist": () => {
+            announceDeliveryStarted();
+            return gate;
+          },
+        },
       });
       const fast = new OutboxDispatcherService({
         store,
         processNames: [processName],
-        leaseDurationMs: 100,
+        leaseDurationMs,
         handlers: { "test.persist": async () => undefined },
       });
 
       const slowRun = slow.runOnce({ now: Date.now() });
-      await sleep(150);
-      const fastReport = await fast.runOnce({ now: Date.now() });
+      await deliveryStarted;
+      // A lease is due again once `leasedUntil <= now`, and `now` is the
+      // caller's, so the rival sees a lapsed lease without the test sleeping
+      // through one.
+      const fastReport = await fast.runOnce({
+        now: Date.now() + leaseDurationMs + 1,
+      });
       expect(fastReport.dispatched).toEqual(["match-000"]);
 
       releaseGate();

--- a/platform/app/src/server/event-sourcing/process-manager/outbox/__tests__/outboxBacklogDrain.integration.test.ts
+++ b/platform/app/src/server/event-sourcing/process-manager/outbox/__tests__/outboxBacklogDrain.integration.test.ts
@@ -207,7 +207,21 @@ describe("outbox backlog drain under slow deliveries", () => {
       });
 
       const slowRun = slow.runOnce({ now: Date.now() });
-      await deliveryStarted;
+      // If the delivery never starts there is no race to observe, and waiting
+      // on the signal alone would spend the whole 60s test timeout finding
+      // that out. Racing the run against the signal turns it into an
+      // assertion that names what went wrong instead. The rejection is
+      // absorbed here on purpose: `slowRun` is awaited again below, which is
+      // where a thrown error belongs.
+      const slowRunSettled = slowRun.then(
+        () => "settled without delivering" as const,
+        () => "settled without delivering" as const,
+      );
+      const raceSetUp = await Promise.race([
+        deliveryStarted.then(() => "delivery started" as const),
+        slowRunSettled,
+      ]);
+      expect(raceSetUp).toBe("delivery started");
       // A lease is due again once `leasedUntil <= now`, and `now` is the
       // caller's, so the rival sees a lapsed lease without the test sleeping
       // through one.

